src/ir/effects.h

```diff
@@ -101,6 +138,8 @@ struct EffectAnalyzer
   bool invalidates(const EffectAnalyzer& other) {
     if ((branches && other.hasSideEffects()) ||
```

Comment:
how about `transfersControlFlow() && other.hasSideEffects()` etc.?

Reply:
Also I think on line 179 we need `transfersControlFlow` too.

Reply:
Done